src/trees/32-byte-code-chunker.md

```diff
@@ -0,0 +1,44 @@
+# 32-byte code-chunker
+
+- [32-byte code-chunker](#32-byte-code-chunker)
+  - [Background reading](#background-reading)
+  - [How does it work?](#how-does-it-work)
+  - [Runtime usage](#runtime-usage)
+  - [Efficiency](#efficiency)
+  - [Implementations](#implementations)
+
+## Background reading
+
+To get a proper background on where this code chunker fits into stateless Ethereum, read the [*Trees*](intro.md) introductory chapter and the [*Code chunking*](data-encoding.md#code-chunking) section of *Data encoding*.
+
+## How does it work?
+
+This approach aims to store code in full 32-byte chunks while efficiently encoding the necessary metadata to validate jump destinations. [This proposal](https://github.com/ipsilon/eof/blob/eof0-dense/spec/eofv0_verkle.md#encode-only-invalid-jumpdests-dense-encoding) contrasts with the [31-byte chunker](31-byte-code-chunker.md) by storing metadata separately rather than prepending it to each chunk.
+
+The goals remain similar:
+
+- Output the code itself as a list of 32-byte blobs (chunks), which will be stored as tree leaves.
+- Provide a mechanism, using separate metadata, for an EVM interpreter to detect if a `JUMP(I)` to any byte offset is valid (i.e., targets a `JUMPDEST` opcode and not PUSHDATA).
+
+Instead of adding metadata to every chunk, the "Dense Encoding" variant method focuses only on potential ambiguities:
+
+- The account bytecode is partitioned into full 32-byte chunks.
+- We identify only the chunks that contain invalid jump destinations. An invalid jump destination is a `0x5b` byte that occurs within the data part of a PUSH instruction.
+- A map `invalid_jumpdests[chunk_index] = first_instruction_offset` is created. first_instruction_offset indicates the offset within the chunk where the first actual instruction begins.
+- This map is then encoded very efficiently using a Variable Length Quantity (VLQ) scheme (specifically, [LEB128](https://en.wikipedia.org/wiki/LEB128)) applied to a combined value representing the distance between invalid chunks and the first_instruction_offset.
+- This densely encoded metadata is stored separately, potentially prepended as a custom table for legacy contracts, i.e., not within the code chunks themselves. EOF contracts can't have invalid jumps since this is validated at deployment time.
```

Comment:
not sure I agree that this is not part of the chunks themselves, somehow they need to be stored in the tree and we might not be able to use the remaining 62 leave of the account header to fit them.

Reply:
Ah, by "not withing the code chunks themselves" I mean as part of the real contract bytecode -- but they must be as part of the code in the tree. I'll clarify better.